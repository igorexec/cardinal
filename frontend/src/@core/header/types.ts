export interface NavigationItem {
  id: string;
  title: string;
  link: string;
}
