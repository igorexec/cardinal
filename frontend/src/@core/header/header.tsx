import React from 'react';
import {NavigationItem} from './types';
import {Navigation} from './navigation';

interface Props {
  navItems: NavigationItem[];
}

export const Header: React.FC<Props> = ({navItems}) => {
  return (
    <header>
      <Navigation items={navItems} />
    </header>
  );
};
