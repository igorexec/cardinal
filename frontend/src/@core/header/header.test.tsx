import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {Header} from './header';

describe('Header', () => {
  let wrapper: ShallowWrapper;
  const props = {navItems: [{id: '1', title: 'title', link: 'link'}]};

  beforeAll(() => {
    wrapper = shallow(<Header {...props} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
