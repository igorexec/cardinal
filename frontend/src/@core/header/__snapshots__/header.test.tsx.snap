// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header>
  <Navigation
    items={
      Array [
        Object {
          "id": "1",
          "link": "link",
          "title": "title",
        },
      ]
    }
  />
</header>
`;
